File: scripts/query_jobs.py
#!/usr/bin/env python3
"""
Job Status Query Script (Python)

Query and display job statuses from the database
Usage: python scripts/query_jobs.py [job_id] [--limit=10] [--status=PENDING]
"""

import os
import sys
import argparse
from datetime import datetime
from dotenv import load_dotenv
from supabase import create_client

load_dotenv()

def get_status_icon(status):
    icons = {
        'PENDING': '⏳',
        'QUEUED': '📋',
        'RUNNING': '🔄',
        'SUCCEEDED': '✅',
        'COMPLETED': '✅',
        'FAILED': '❌',
        'CANCELLED': '🚫'
    }
    return icons.get(status, '❓')

def get_progress_bar(progress, width=20):
    if progress is None:
        progress = 0
    filled = int((progress / 100) * width)
    empty = width - filled
    return '█' * filled + '░' * empty

def format_datetime(dt_str):
    try:
        dt = datetime.fromisoformat(dt_str.replace('Z', '+00:00'))
        return dt.strftime('%Y-%m-%d %H:%M:%S')
    except:
        return dt_str

def query_jobs(job_id=None, limit=10, status_filter=None):
    print('📋 Job Status Query (Python)')
    print('=============================')
    
    try:
        supabase_url = os.getenv("SUPABASE_URL")
        supabase_service_key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
        
        if not all([supabase_url, supabase_service_key]):
            print("❌ Missing Supabase configuration")
            return
        
        supabase = create_client(supabase_url, supabase_service_key)
        
        if job_id:
            # Query specific job
            print(f"🔍 Querying specific job: {job_id}")
            response = supabase.table("jobs").select("*").eq("id", job_id).execute()
        else:
            # Query multiple jobs
            print(f"🔍 Querying recent jobs (limit: {limit})")
            query = supabase.table("jobs").select("*")
            
            if status_filter:
                print(f"   Status filter: {status_filter}")
                query = query.eq("status", status_filter)
            
            response = query.order("created_at", desc=True).limit(limit).execute()
        
        jobs = response.data
        
        if not jobs:
            print("📭 No jobs found")
            return
        
        print(f"\n📊 Found {len(jobs)} job(s):")
        print('=' * 80)
        
        for i, job in enumerate(jobs, 1):
            status_icon = get_status_icon(job['status'])
            progress_bar = get_progress_bar(job.get('progress', 0))
            
            print(f"\n{i}. Job {job['id']}")
            print(f"   Status: {status_icon} {job['status']}")
            print(f"   Progress: {progress_bar} {job.get('progress', 0)}%")
            print(f"   User: {job['user_id']}")
            print(f"   Created: {format_datetime(job['created_at'])}")
            print(f"   Updated: {format_datetime(job['updated_at'])}")
            
            if job.get('youtube_url'):
                print(f"   YouTube: {job['youtube_url']}")
            
            if job.get('error'):
                print(f"   Error: {job['error']}")
        
        # Summary statistics
        print('\n📈 Status Summary:')
        status_counts = {}
        for job in jobs:
            status = job['status']
            status_counts[status] = status_counts.get(status, 0) + 1
        
        for status, count in status_counts.items():
            icon = get_status_icon(status)
            print(f"   {icon} {status}: {count}")
        
    except Exception as e:
        print(f"❌ Query failed: {e}")
        import traceback
        traceback.print_exc()

def main():
    parser = argparse.ArgumentParser(description='Query job statuses from database')
    parser.add_argument('job_id', nargs='?', help='Specific job ID to query')
    parser.add_argument('--limit', type=int, default=10, help='Limit number of results (default: 10)')
    parser.add_argument('--status', help='Filter by status (PENDING, RUNNING, SUCCEEDED, etc.)')
    
    args = parser.parse_args()
    
    query_jobs(args.job_id, args.limit, args.status)

if __name__ == "__main__":
    main()

File: scripts/query-jobs.js
#!/usr/bin/env node

/**
 * Job Status Query Script
 * 
 * Query and display job statuses from the database
 * Usage: node scripts/query-jobs.js [job_id] [--limit=10] [--status=PENDING]
 */

const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const supabase = createClient(
    process.env.SUPABASE_URL,
    process.env.SUPABASE_SERVICE_ROLE_KEY
);

async function queryJobs() {
    console.log('📋 Job Status Query');
    console.log('===================');
    
    try {
        // Parse command line arguments
        const args = process.argv.slice(2);
        const jobId = args.find(arg => !arg.startsWith('--'));
        const limitArg = args.find(arg => arg.startsWith('--limit='));
        const statusArg = args.find(arg => arg.startsWith('--status='));
        
        const limit = limitArg ? parseInt(limitArg.split('=')[1]) : 10;
        const statusFilter = statusArg ? statusArg.split('=')[1] : null;
        
        let query = supabase.from('jobs').select('*');
        
        if (jobId) {
            // Query specific job
            console.log(`🔍 Querying specific job: ${jobId}`);
            query = query.eq('id', jobId);
        } else {
            // Query multiple jobs
            console.log(`🔍 Querying recent jobs (limit: ${limit})`);
            if (statusFilter) {
                console.log(`   Status filter: ${statusFilter}`);
                query = query.eq('status', statusFilter);
            }
            query = query.order('created_at', { ascending: false }).limit(limit);
        }
        
        const { data: jobs, error } = await query;
        
        if (error) {
            console.error('❌ Query failed:', error.message);
            return;
        }
        
        if (!jobs || jobs.length === 0) {
            console.log('📭 No jobs found');
            return;
        }
        
        console.log(`\n📊 Found ${jobs.length} job(s):`);
        console.log(''.padEnd(80, '='));
        
        jobs.forEach((job, index) => {
            const statusIcon = getStatusIcon(job.status);
            const progressBar = getProgressBar(job.progress || 0);
            
            console.log(`\n${index + 1}. Job ${job.id}`);
            console.log(`   Status: ${statusIcon} ${job.status}`);
            console.log(`   Progress: ${progressBar} ${job.progress || 0}%`);
            console.log(`   User: ${job.user_id}`);
            console.log(`   Created: ${new Date(job.created_at).toLocaleString()}`);
            console.log(`   Updated: ${new Date(job.updated_at).toLocaleString()}`);
            
            if (job.youtube_url) {
                console.log(`   YouTube: ${job.youtube_url}`);
            }
            
            if (job.error) {
                console.log(`   Error: ${job.error}`);
            }
        });
        
        // Summary statistics
        console.log('\n📈 Status Summary:');
        const statusCounts = jobs.reduce((acc, job) => {
            acc[job.status] = (acc[job.status] || 0) + 1;
            return acc;
        }, {});
        
        Object.entries(statusCounts).forEach(([status, count]) => {
            const icon = getStatusIcon(status);
            console.log(`   ${icon} ${status}: ${count}`);
        });
        
    } catch (error) {
        console.error('❌ Query failed:', error.message);
    }
}

function getStatusIcon(status) {
    const icons = {
        'PENDING': '⏳',
        'QUEUED': '📋',
        'RUNNING': '🔄',
        'SUCCEEDED': '✅',
        'COMPLETED': '✅',
        'FAILED': '❌',
        'CANCELLED': '🚫'
    };
    return icons[status] || '❓';
}

function getProgressBar(progress) {
    const width = 20;
    const filled = Math.round((progress / 100) * width);
    const empty = width - filled;
    return '█'.repeat(filled) + '░'.repeat(empty);
}

function printUsage() {
    console.log('Usage: node scripts/query-jobs.js [job_id] [options]');
    console.log('');
    console.log('Options:');
    console.log('  --limit=N     Limit number of results (default: 10)');
    console.log('  --status=S    Filter by status (PENDING, RUNNING, SUCCEEDED, etc.)');
    console.log('');
    console.log('Examples:');
    console.log('  node scripts/query-jobs.js                           # Recent 10 jobs');
    console.log('  node scripts/query-jobs.js --limit=5                 # Recent 5 jobs');
    console.log('  node scripts/query-jobs.js --status=SUCCEEDED        # All succeeded jobs');
    console.log('  node scripts/query-jobs.js fa1fb2fa-34dd-4fe0-b718-52b488a45769  # Specific job');
}

if (require.main === module) {
    if (process.argv.includes('--help') || process.argv.includes('-h')) {
        printUsage();
    } else {
        queryJobs();
    }
}

module.exports = { queryJobs };
